### PR TITLE
Initiate checkout on log in

### PR DESCRIPTION
**Pull request by danac-gs**
There are 2 ways of login, through 2 separate controllers, Account and Ajax. The code redirects to shopping cart page if the cart exists after user logs in through any of them. The logic is appended to the default controller action in form of a plugin, overriding the default result by adding a redirect.

### CODE REVIEW

File: Plugin/LoginPlugin.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            $response['redirectUrl'] = self::$shoppingCartPath;
+
+            // Set the flag in session to auto-open Bolt checkout on redirected (shopping cart) page
+            $this->setBoltOpenCheckoutFlag();
+
+            $resultJson = $this->resultJsonFactory->create();
+            return $resultJson->setData($response);
+
+        } catch (\Exception $e) {
+            // On any exception pass the original result through
```
Comment (daisy1754) [inline Plugin/LoginPlugin.php]: let's bugsnag

File: Plugin/LoginPluginTrait.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+     * @return bool
+     */
+    private function hasCart() {
+        return $this->checkoutSession->hasQuote() && count($this->checkoutSession->getQuote()->getAllVisibleItems()) > 0;
+    }
+
+    /**
+     * @return bool
+     */
+    private function allowRedirect() {
```
Comment (daisy1754) [inline Plugin/LoginPluginTrait.php]: shouldRecirectToCartPage

**Comment by YevhenShBolt** [comment]
Sorry @danac-gs, i posted the wrong review above.

**Comment by danac-gs** [comment]
@daisy1754 @mccottry 
The code is refactored in a way that a parent abstract class is used instead of a trait.
Checkout auto-opens after login only if click on a bolt button triggered user authentication.

**Pull request merged**